### PR TITLE
Add submenu unique

### LINKED ISSUE #131: Issues with multiple submenus with same id
We currently can add multiple submenus with the same "id"
Example:
```php
$menu = (new CliMenuBuilder)
    ->addSubMenu('Options')
        ->addItem(...)
        ->end()
    ->addSubMenu('Options')
        ->addItem(...)
        ->end()
    ->end()
```
Both of them work but only the second one has its parent set and can "Go Back" to the main menu.

We either need to prevent adding a menu with the same ID or, even better, handle it differently (maybe like the `uniqid` that's used with split items ?).

### PR DESCRIPTION
**Pull request by AydinHassan**
Fixes #131.

### CODE REVIEW

File: src/Builder/CliMenuBuilder.php [hunk 1]
```diff
@@ -117,8 +117,11 @@ public function addAsciiArt(string $art, string $position = AsciiArtItem::POSITI
      */
     public function addSplitItem() : SplitItemBuilder
     {
-        $this->menuItems[] = $id = uniqid('splititem-placeholder-', true);
-        
+        $this->menuItems[] = [
+            'type' => 'splititem-placeholder',
+            'id'   => $id = uniqid('', true),
```
Comment (Lynesth) [inline src/Builder/CliMenuBuilder.php]: Should we allow optionnal user defined id ?
Do we have any interest in allowing to getSplitItem() like we do with getSubMenu() ?

File: src/Builder/BuilderUtils.php [hunk 2]
```diff
@@ -62,10 +62,20 @@ public function addLineBreak(string $breakChar = ' ', int $lines = 1) : self
      * Add a submenu with a name. The name will be displayed as the item text
      * in the parent menu.
      */
-    public function addSubMenu(string $name, CliMenuBuilder $subMenuBuilder = null) : Builder
+    public function addSubMenu(string $id, string $text, CliMenuBuilder $subMenuBuilder = null) : Builder
     {
-        $this->menuItems[]  = $id = 'submenu-placeholder-' . $name;
-
+        if (isset($this->subMenuBuilders[$id])) {
```
Comment (Lynesth) [inline src/Builder/BuilderUtils.php]: I guess if you **really** don't want to break BC too much on this, you could set $text to be optionnal (null by default) and then do something like to allow mismatched arguments:
```php
if ($text instanceof CliMenuBuilder) {
    $subMenuBuilder = $text;
    $text = $id;
} elseif (null === $text) {
    $text = $id;
}
```

Not too inclined to do so but that's just a remark.

File: src/Builder/CliMenuBuilder.php (hunk 1 above)
Comment (AydinHassan) [reply src/Builder/CliMenuBuilder.php]: No - I don't want to add an API for something there is no concrete use case for. Same reason I'm in this mess in the first place.

File: src/Builder/BuilderUtils.php (hunk 2 above)
Comment (AydinHassan) [reply src/Builder/BuilderUtils.php]: No let's just keep it as it is here.

File: UPGRADE.md [hunk 3]
```diff
@@ -11,4 +11,7 @@ or backwards compatibility (BC) breakages occur.
   `PhpSchool\CliMenu\Builder\CliMenuBuilder` instead. The old class has been aliased for now, but will be removed in 
   `3.1`.
 * Removed `PhpSchool\CliMenu\Terminal` namespace, the code has been migrated to the `php-school/terminal` package and is 
-  largely modified.
\ No newline at end of file
+  largely modified.
+* Method `addSubMenu` in '\PhpSchool\CliMenu\Builder\CliMenuBuilder' has an additional required parameter
+  added at the beginning of the parameter list. It must be a unique ID for the submenu. The method can be found in the trait
+  `\PhpSchool\CliMenu\Builder\BuilderUtils`.
```
[the remark this answers is not in the document]
Comment (AydinHassan) [reply UPGRADE.md]: @Lynesth fixed.

**Comment by AydinHassan** [comment]
Not needed anymore.

**Pull request closed without merging**